{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "bd5fb3a8",
   "metadata": {},
   "source": [
    "### Extract Eur and AFR samples from 1000G"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3ca91f3a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Populating the interactive namespace from numpy and matplotlib\n"
     ]
    }
   ],
   "source": [
    "# Import libraries\n",
    "%pylab inline\n",
    "import pandas as pd\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6e034ded",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_1.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_1\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_1-temporary.bed + EUR_1-temporary.bim + EUR_1-temporary.fam written.\n",
      "6468094 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_1.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.99987.\n",
      "6468094 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_1.bed + EUR_1.bim + EUR_1.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_2.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_2\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr2.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_2-temporary.bed + EUR_2-temporary.bim + EUR_2-temporary.fam written.\n",
      "7081600 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_2.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999873.\n",
      "7081600 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_2.bed + EUR_2.bim + EUR_2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_3.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_3\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr3.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_3-temporary.bed + EUR_3-temporary.bim + EUR_3-temporary.fam written.\n",
      "5832276 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_3.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999863.\n",
      "5832276 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_3.bed + EUR_3.bim + EUR_3.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_4.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_4\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr4.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_4-temporary.bed + EUR_4-temporary.bim + EUR_4-temporary.fam written.\n",
      "5732585 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_4.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999852.\n",
      "5732585 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_4.bed + EUR_4.bim + EUR_4.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_5.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_5\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr5.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_5-temporary.bed + EUR_5-temporary.bim + EUR_5-temporary.fam written.\n",
      "5265763 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_5.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999868.\n",
      "5265763 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_5.bed + EUR_5.bim + EUR_5.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_6.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_6\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr6.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_6-temporary.bed + EUR_6-temporary.bim + EUR_6-temporary.fam written.\n",
      "5024119 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_6.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999837.\n",
      "5024119 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_6.bed + EUR_6.bim + EUR_6.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_7.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_7\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr7.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_7-temporary.bed + EUR_7-temporary.bim + EUR_7-temporary.fam written.\n",
      "4716715 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_7.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999866.\n",
      "4716715 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_7.bed + EUR_7.bim + EUR_7.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_8.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_8\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr8.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_8-temporary.bed + EUR_8-temporary.bim + EUR_8-temporary.fam written.\n",
      "4597105 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_8.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999872.\n",
      "4597105 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_8.bed + EUR_8.bim + EUR_8.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_9.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_9\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr9.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_9-temporary.bed + EUR_9-temporary.bim + EUR_9-temporary.fam written.\n",
      "3560687 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_9.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999866.\n",
      "3560687 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_9.bed + EUR_9.bim + EUR_9.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_10.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_10\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr10.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_10-temporary.bed + EUR_10-temporary.bim + EUR_10-temporary.fam\n",
      "written.\n",
      "3992219 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_10.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.99986.\n",
      "3992219 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_10.bed + EUR_10.bim + EUR_10.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_11.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_11\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr11.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_11-temporary.bed + EUR_11-temporary.bim + EUR_11-temporary.fam\n",
      "written.\n",
      "4045628 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_11.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999862.\n",
      "4045628 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_11.bed + EUR_11.bim + EUR_11.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_12.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_12\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr12.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--vcf: EUR_12-temporary.bed + EUR_12-temporary.bim + EUR_12-temporary.fam\n",
      "written.\n",
      "3868428 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_12.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999855.\n",
      "3868428 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_12.bed + EUR_12.bim + EUR_12.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_13.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_13\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr13.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_13-temporary.bed + EUR_13-temporary.bim + EUR_13-temporary.fam\n",
      "written.\n",
      "2857916 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_13.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999842.\n",
      "2857916 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_13.bed + EUR_13.bim + EUR_13.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_14.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_14\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr14.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_14-temporary.bed + EUR_14-temporary.bim + EUR_14-temporary.fam\n",
      "written.\n",
      "2655067 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_14.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999869.\n",
      "2655067 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_14.bed + EUR_14.bim + EUR_14.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_15.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_15\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_15-temporary.bed + EUR_15-temporary.bim + EUR_15-temporary.fam\n",
      "written.\n",
      "2424689 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_15.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999863.\n",
      "2424689 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_15.bed + EUR_15.bim + EUR_15.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_16.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_16\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr16.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_16-temporary.bed + EUR_16-temporary.bim + EUR_16-temporary.fam\n",
      "written.\n",
      "2697949 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_16.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999882.\n",
      "2697949 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_16.bed + EUR_16.bim + EUR_16.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_17.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_17\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr17.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_17-temporary.bed + EUR_17-temporary.bim + EUR_17-temporary.fam\n",
      "written.\n",
      "2329288 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_17.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999877.\n",
      "2329288 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--make-bed to EUR_17.bed + EUR_17.bim + EUR_17.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_18.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_18\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr18.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_18-temporary.bed + EUR_18-temporary.bim + EUR_18-temporary.fam\n",
      "written.\n",
      "2267185 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_18.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999855.\n",
      "2267185 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_18.bed + EUR_18.bim + EUR_18.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_19.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_19\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr19.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_19-temporary.bed + EUR_19-temporary.bim + EUR_19-temporary.fam\n",
      "written.\n",
      "1832506 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_19.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999863.\n",
      "1832506 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_19.bed + EUR_19.bim + EUR_19.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_20.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_20\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr20.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_20-temporary.bed + EUR_20-temporary.bim + EUR_20-temporary.fam\n",
      "written.\n",
      "1812841 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_20.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999882.\n",
      "1812841 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_20.bed + EUR_20.bim + EUR_20.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_21.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_21\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr21.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_21-temporary.bed + EUR_21-temporary.bim + EUR_21-temporary.fam\n",
      "written.\n",
      "1105538 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_21.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999831.\n",
      "1105538 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_21.bed + EUR_21.bim + EUR_21.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_22.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/EUR_sample.txt\n",
      "  --make-bed\n",
      "  --out EUR_22\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr22.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: EUR_22-temporary.bed + EUR_22-temporary.bim + EUR_22-temporary.fam\n",
      "written.\n",
      "1103547 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_22.nosex .\n",
      "--keep: 502 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999881.\n",
      "1103547 variants and 502 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to EUR_22.bed + EUR_22.bim + EUR_22.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_1.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_1\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_1-temporary.bed + AFR_1-temporary.bim + AFR_1-temporary.fam written.\n",
      "6468094 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_1.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.99977.\n",
      "6468094 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_1.bed + AFR_1.bim + AFR_1.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_2.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_2\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr2.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_2-temporary.bed + AFR_2-temporary.bim + AFR_2-temporary.fam written.\n",
      "7081600 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_2.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999762.\n",
      "7081600 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_2.bed + AFR_2.bim + AFR_2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_3.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_3\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr3.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_3-temporary.bed + AFR_3-temporary.bim + AFR_3-temporary.fam written.\n",
      "5832276 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_3.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999751.\n",
      "5832276 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_3.bed + AFR_3.bim + AFR_3.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_4.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_4\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr4.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_4-temporary.bed + AFR_4-temporary.bim + AFR_4-temporary.fam written.\n",
      "5732585 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_4.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999735.\n",
      "5732585 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_4.bed + AFR_4.bim + AFR_4.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_5.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_5\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr5.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_5-temporary.bed + AFR_5-temporary.bim + AFR_5-temporary.fam written.\n",
      "5265763 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_5.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999752.\n",
      "5265763 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_5.bed + AFR_5.bim + AFR_5.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_6.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_6\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr6.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_6-temporary.bed + AFR_6-temporary.bim + AFR_6-temporary.fam written.\n",
      "5024119 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_6.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999715.\n",
      "5024119 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_6.bed + AFR_6.bim + AFR_6.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_7.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_7\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr7.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--vcf: AFR_7-temporary.bed + AFR_7-temporary.bim + AFR_7-temporary.fam written.\n",
      "4716715 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_7.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999753.\n",
      "4716715 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_7.bed + AFR_7.bim + AFR_7.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_8.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_8\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr8.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_8-temporary.bed + AFR_8-temporary.bim + AFR_8-temporary.fam written.\n",
      "4597105 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_8.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.99975.\n",
      "4597105 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_8.bed + AFR_8.bim + AFR_8.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_9.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_9\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr9.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_9-temporary.bed + AFR_9-temporary.bim + AFR_9-temporary.fam written.\n",
      "3560687 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_9.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999763.\n",
      "3560687 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_9.bed + AFR_9.bim + AFR_9.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_10.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_10\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr10.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_10-temporary.bed + AFR_10-temporary.bim + AFR_10-temporary.fam\n",
      "written.\n",
      "3992219 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_10.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999752.\n",
      "3992219 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_10.bed + AFR_10.bim + AFR_10.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_11.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_11\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr11.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_11-temporary.bed + AFR_11-temporary.bim + AFR_11-temporary.fam\n",
      "written.\n",
      "4045628 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_11.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999755.\n",
      "4045628 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_11.bed + AFR_11.bim + AFR_11.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_12.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_12\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr12.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_12-temporary.bed + AFR_12-temporary.bim + AFR_12-temporary.fam\n",
      "written.\n",
      "3868428 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_12.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999743.\n",
      "3868428 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_12.bed + AFR_12.bim + AFR_12.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_13.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_13\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr13.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--vcf: AFR_13-temporary.bed + AFR_13-temporary.bim + AFR_13-temporary.fam\n",
      "written.\n",
      "2857916 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_13.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999737.\n",
      "2857916 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_13.bed + AFR_13.bim + AFR_13.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_14.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_14\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr14.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_14-temporary.bed + AFR_14-temporary.bim + AFR_14-temporary.fam\n",
      "written.\n",
      "2655067 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_14.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999757.\n",
      "2655067 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_14.bed + AFR_14.bim + AFR_14.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_15.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_15\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_15-temporary.bed + AFR_15-temporary.bim + AFR_15-temporary.fam\n",
      "written.\n",
      "2424689 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_15.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999754.\n",
      "2424689 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_15.bed + AFR_15.bim + AFR_15.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_16.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_16\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr16.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_16-temporary.bed + AFR_16-temporary.bim + AFR_16-temporary.fam\n",
      "written.\n",
      "2697949 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_16.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999765.\n",
      "2697949 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_16.bed + AFR_16.bim + AFR_16.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_17.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_17\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr17.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_17-temporary.bed + AFR_17-temporary.bim + AFR_17-temporary.fam\n",
      "written.\n",
      "2329288 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_17.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999752.\n",
      "2329288 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_17.bed + AFR_17.bim + AFR_17.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_18.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_18\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr18.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_18-temporary.bed + AFR_18-temporary.bim + AFR_18-temporary.fam\n",
      "written.\n",
      "2267185 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_18.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999742.\n",
      "2267185 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--make-bed to AFR_18.bed + AFR_18.bim + AFR_18.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_19.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_19\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr19.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_19-temporary.bed + AFR_19-temporary.bim + AFR_19-temporary.fam\n",
      "written.\n",
      "1832506 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_19.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999729.\n",
      "1832506 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_19.bed + AFR_19.bim + AFR_19.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_20.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_20\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr20.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_20-temporary.bed + AFR_20-temporary.bim + AFR_20-temporary.fam\n",
      "written.\n",
      "1812841 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_20.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999764.\n",
      "1812841 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_20.bed + AFR_20.bim + AFR_20.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_21.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_21\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr21.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_21-temporary.bed + AFR_21-temporary.bim + AFR_21-temporary.fam\n",
      "written.\n",
      "1105538 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_21.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.99968.\n",
      "1105538 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_21.bed + AFR_21.bim + AFR_21.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR_22.log.\n",
      "Options in effect:\n",
      "  --keep /home/hex002/ps2-wi24/AFR_sample.txt\n",
      "  --make-bed\n",
      "  --out AFR_22\n",
      "  --vcf /home/hex002/public/1000Genomes/ALL.chr22.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "--vcf: AFR_22-temporary.bed + AFR_22-temporary.bim + AFR_22-temporary.fam\n",
      "written.\n",
      "1103547 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to AFR_22.nosex .\n",
      "--keep: 661 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999745.\n",
      "1103547 variants and 661 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to AFR_22.bed + AFR_22.bim + AFR_22.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "sample = pd.read_csv('~/public/1000Genomes/igsr_samples.tsv',sep = '\\t')\n",
    "for anc in ['EUR','AFR']:\n",
    "    sub_sample = sample[sample['Superpopulation code'] == anc][['Sample name']]\n",
    "    sub_sample['IID'] = sub_sample['Sample name']\n",
    "    sub_sample.to_csv(f'{anc}_sample.txt',sep = '\\t',index = False,header = False)\n",
    "    for i in range(1,23):\n",
    "        os.system(f\"plink --vcf ~/public/1000Genomes/ALL.chr{i}.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz \\\n",
    "                    --keep /home/hex002/ps2-wi24/{anc}_sample.txt \\\n",
    "                    --make-bed \\\n",
    "                    --out {anc}_{i}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "222a0210",
   "metadata": {},
   "outputs": [],
   "source": [
    "sample = pd.read_csv('~/public/1000Genomes/igsr_samples.tsv',sep = '\\t')\n",
    "for anc in ['EUR','AFR']:\n",
    "    sub_sample = sample[sample['Superpopulation code'] == anc][['Sample name']]\n",
    "    sub_sample['IID'] = sub_sample['Sample name']\n",
    "    sub_sample.to_csv(f'{anc}_sample.txt',sep = '\\t',index = False,header = False)\n",
    "    for i in range(1,23):\n",
    "        os.system(f\"plink --vcf ~/public/1000Genomes/ALL.chr{i}.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz \\\n",
    "                    --keep /home/hex002/ps2-wi24/{anc}_sample.txt \\\n",
    "                    --make-bed \\\n",
    "                    --out {anc}_{i}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21bc6e35",
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in {1..22}\n",
    "do\n",
    "echo EUR_${i}_filtered >> EUR_mergelist.txt\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8c7b8ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "plink --merge-list EUR_mergelist.txt  --make-bed --out 1000G_EUR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2fb10c6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "plink --merge-list AFR_mergelist.txt --make-bed --out 1000G_AFR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "47668179",
   "metadata": {},
   "outputs": [],
   "source": [
    "./gcta/gcta-1.94.1 --bfile EUR_22 --autosome --maf 0.05 --make-grm --out EUR_22 --thread-num 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3bfa8ecc",
   "metadata": {},
   "outputs": [],
   "source": [
    "./gcta/gcta-1.94.1 --grm EUR_22 --pheno EUR_22.phen --reml --out EUR_22 --thread-num 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "abba0537",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda create -n heels python=3.10.12 pip numpy pandas scipy pandas-plink joblib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f42dfb66",
   "metadata": {},
   "outputs": [],
   "source": [
    "ld = np.load('ukb_chr22.npz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f5a28891",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:05:18 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_1 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_1 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_1.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_1.bim]...\n",
      "6468094 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 6468094 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_1.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 3.7 sec\n",
      "6468094 SNPs have been processed.\n",
      "  Used 534817 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_1.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_1.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:05:30 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 11.45 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:05:30 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_2 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_2 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_2.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_2.bim]...\n",
      "7081600 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 7081600 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_2.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 4.4 sec\n",
      "7081600 SNPs have been processed.\n",
      "  Used 578116 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_2.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_2.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:05:43 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 12.78 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:05:43 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_3 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_3 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_3.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_3.bim]...\n",
      "5832276 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 5832276 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_3.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 3.3 sec\n",
      "5832276 SNPs have been processed.\n",
      "  Used 492426 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_3.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_3.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:05:53 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 10.18 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:05:53 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_4 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_4 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_4.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_4.bim]...\n",
      "5732585 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 5732585 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_4.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 3.7 sec\n",
      "5732585 SNPs have been processed.\n",
      "  Used 508382 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_4.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_4.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:06:04 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 10.59 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:06:04 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_5 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_5 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_5.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_5.bim]...\n",
      "5265763 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 5265763 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_5.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 3.6 sec\n",
      "5265763 SNPs have been processed.\n",
      "  Used 437745 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_5.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_5.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:06:14 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 10.05 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:06:14 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_6 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_6 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_6.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_6.bim]...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5024119 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 5024119 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_6.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 3.3 sec\n",
      "5024119 SNPs have been processed.\n",
      "  Used 479079 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_6.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_6.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:06:23 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 9.71 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:06:23 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_7 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_7 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_7.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_7.bim]...\n",
      "4716715 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 4716715 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_7.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 3.1 sec\n",
      "4716715 SNPs have been processed.\n",
      "  Used 413283 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_7.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_7.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:06:32 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 9.05 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:06:32 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_8 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_8 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_8.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_8.bim]...\n",
      "4597105 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 4597105 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_8.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 3.0 sec\n",
      "4597105 SNPs have been processed.\n",
      "  Used 377954 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_8.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_8.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:06:41 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 8.77 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:06:41 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_9 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_9 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_9.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_9.bim]...\n",
      "3560687 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 3560687 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_9.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 2.3 sec\n",
      "3560687 SNPs have been processed.\n",
      "  Used 301258 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_9.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_9.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:06:48 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 6.59 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:06:48 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_10 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_10 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_10.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_10.bim]...\n",
      "3992219 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 3992219 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_10.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 2.7 sec\n",
      "3992219 SNPs have been processed.\n",
      "  Used 359414 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_10.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_10.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:06:55 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 7.47 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:06:55 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_11 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_11 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_11.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_11.bim]...\n",
      "4045628 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 4045628 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_11.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 2.1 sec\n",
      "4045628 SNPs have been processed.\n",
      "  Used 349403 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_11.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_11.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:07:02 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 6.93 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:07:02 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_12 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_12 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_12.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_12.bim]...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3868428 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 3868428 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_12.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 2.8 sec\n",
      "3868428 SNPs have been processed.\n",
      "  Used 336703 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_12.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_12.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:07:10 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 7.46 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:07:10 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_13 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_13 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_13.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_13.bim]...\n",
      "2857916 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 2857916 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_13.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 1.6 sec\n",
      "2857916 SNPs have been processed.\n",
      "  Used 258165 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_13.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_13.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:07:15 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 4.99 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:07:15 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_14 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_14 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_14.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_14.bim]...\n",
      "2655067 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 2655067 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_14.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 1.4 sec\n",
      "2655067 SNPs have been processed.\n",
      "  Used 228941 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_14.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_14.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:07:19 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 4.64 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:07:19 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_15 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_15 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_15.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_15.bim]...\n",
      "2424689 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 2424689 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_15.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 1.3 sec\n",
      "2424689 SNPs have been processed.\n",
      "  Used 199709 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_15.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_15.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:07:23 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 4.21 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:07:23 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_16 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_16 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_16.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_16.bim]...\n",
      "2697949 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 2697949 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_16.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 1.5 sec\n",
      "2697949 SNPs have been processed.\n",
      "  Used 213534 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_16.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_16.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:07:28 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 4.84 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:07:28 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_17 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_17 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_17.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_17.bim]...\n",
      "2329288 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 2329288 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_17.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 1.2 sec\n",
      "2329288 SNPs have been processed.\n",
      "  Used 192179 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_17.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_17.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:07:32 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 3.98 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:07:32 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_18 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_18 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_18.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_18.bim]...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2267185 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 2267185 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_18.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 1.7 sec\n",
      "2267185 SNPs have been processed.\n",
      "  Used 197235 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_18.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_18.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:07:37 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 4.51 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:07:37 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_19 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_19 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_19.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_19.bim]...\n",
      "1832506 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 1832506 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_19.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 1.4 sec\n",
      "1832506 SNPs have been processed.\n",
      "  Used 167722 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_19.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_19.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:07:40 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 3.46 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:07:40 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_20 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_20 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_20.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_20.bim]...\n",
      "1812841 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 1812841 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_20.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 1.3 sec\n",
      "1812841 SNPs have been processed.\n",
      "  Used 153290 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_20.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_20.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:07:44 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 3.49 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:07:44 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_21 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_21 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_21.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_21.bim]...\n",
      "1105538 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 1105538 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_21.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 0.7 sec\n",
      "1105538 SNPs have been processed.\n",
      "  Used 101877 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_21.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_21.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:07:46 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 2.04 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:07:46 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Options: \n",
      " \n",
      "--bfile EUR_22 \n",
      "--autosome \n",
      "--maf 0.05 \n",
      "--make-grm \n",
      "--out EUR_22 \n",
      "--thread-num 5 \n",
      "\n",
      "The program will be running with up to 5 threads.\n",
      "Note: GRM is computed using the SNPs on the autosomes.\n",
      "Reading PLINK FAM file from [EUR_22.fam]...\n",
      "502 individuals to be included from FAM file.\n",
      "502 individuals to be included. 0 males, 0 females, 502 unknown.\n",
      "Reading PLINK BIM file from [EUR_22.bim]...\n",
      "1103547 SNPs to be included from BIM file(s).\n",
      "Threshold to filter variants: MAF > 0.050000.\n",
      "Computing the genetic relationship matrix (GRM) v2 ...\n",
      "Subset 1/1, no. subject 1-502\n",
      "  502 samples, 1103547 markers, 126253 GRM elements\n",
      "IDs for the GRM file have been saved in the file [EUR_22.grm.id]\n",
      "Computing GRM...\n",
      "  100% finished in 0.8 sec\n",
      "1103547 SNPs have been processed.\n",
      "  Used 98975 valid SNPs.\n",
      "The GRM computation is completed.\n",
      "Saving GRM...\n",
      "GRM has been saved in the file [EUR_22.grm.bin]\n",
      "Number of SNPs in each pair of individuals has been saved in the file [EUR_22.grm.N.bin]\n",
      "\n",
      "Analysis finished at 03:07:48 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 2.07 sec.\n"
     ]
    }
   ],
   "source": [
    "for anc in ['EUR']:\n",
    "    for i in range(1,23):\n",
    "        os.system(f\"./gcta/gcta-1.94.1 --bfile EUR_{i} --autosome --maf 0.05 --make-grm --out EUR_{i} --thread-num 5\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "90908c64",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:51 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_1\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_1\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_1.grm.id].\n",
      "502 IDs are read from [EUR_1.grm.id].\n",
      "Reading the GRM from [EUR_1.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_1.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values:  0.45687 0.468325\n",
      "logL: -231.867\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-231.77\t0.35302\t0.56300\t\n",
      "2\t-230.73\t0.17011\t0.73720\t\n",
      "3\t-230.25\t0.18837\t0.72094\t\n",
      "4\t-230.20\t0.22459\t0.68936\t\n",
      "5\t-230.17\t0.21558\t0.69832\t\n",
      "6\t-230.17\t0.21818\t0.69582\t\n",
      "7\t-230.17\t0.21745\t0.69653\t\n",
      "8\t-230.17\t0.21766\t0.69633\t\n",
      "9\t-230.17\t0.21760\t0.69638\t\n",
      "10\t-230.17\t0.21762\t0.69637\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.217617\t0.138533\n",
      "V(e)\t0.696367\t0.140631\n",
      "Vp\t0.913984\t0.057906\n",
      "V(G)/Vp\t0.238097\t0.149708\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "1.919126e-02\t-1.780770e-02\t\n",
      "-1.780770e-02\t1.977722e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_1.hsq].\n",
      "\n",
      "Analysis finished at 03:08:51 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.17 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:51 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_2\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_2\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_2.grm.id].\n",
      "502 IDs are read from [EUR_2.grm.id].\n",
      "Reading the GRM from [EUR_2.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_2.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.457505 0.464896\n",
      "logL: -230.443\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-230.40\t0.38064\t0.53584\t\n",
      "2\t-229.88\t0.23221\t0.67659\t\n",
      "3\t-229.55\t0.26253\t0.65025\t\n",
      "4\t-229.53\t0.25804\t0.65474\t\n",
      "5\t-229.53\t0.25884\t0.65397\t\n",
      "6\t-229.53\t0.25870\t0.65410\t\n",
      "7\t-229.53\t0.25872\t0.65408\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.258725\t0.149225\n",
      "V(e)\t0.654078\t0.148843\n",
      "Vp\t0.912803\t0.057919\n",
      "V(G)/Vp\t0.283440\t0.160839\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "2.226805e-02\t-2.053380e-02\t\n",
      "-2.053380e-02\t2.215420e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_2.hsq].\n",
      "\n",
      "Analysis finished at 03:08:51 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.11 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:51 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_3\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_3\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_3.grm.id].\n",
      "502 IDs are read from [EUR_3.grm.id].\n",
      "Reading the GRM from [EUR_3.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_3.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.456955 0.468785\n",
      "logL: -231.467\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-231.37\t0.36136\t0.55500\t\n",
      "2\t-230.37\t0.18682\t0.71951\t\n",
      "3\t-229.82\t0.23197\t0.68097\t\n",
      "4\t-229.77\t0.22402\t0.68907\t\n",
      "5\t-229.77\t0.22583\t0.68735\t\n",
      "6\t-229.77\t0.22543\t0.68773\t\n",
      "7\t-229.77\t0.22552\t0.68764\t\n",
      "8\t-229.77\t0.22550\t0.68766\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.225501\t0.130581\n",
      "V(e)\t0.687663\t0.132246\n",
      "Vp\t0.913165\t0.057903\n",
      "V(G)/Vp\t0.246945\t0.140821\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "1.705147e-02\t-1.559385e-02\t\n",
      "-1.559385e-02\t1.748903e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_3.hsq].\n",
      "\n",
      "Analysis finished at 03:08:51 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.16 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:51 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_4\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_4\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_4.grm.id].\n",
      "502 IDs are read from [EUR_4.grm.id].\n",
      "Reading the GRM from [EUR_4.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_4.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.457103 0.465954\n",
      "logL: -229.886\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-229.83\t0.38850\t0.52829\t\n",
      "2\t-229.29\t0.25583\t0.65272\t\n",
      "3\t-228.95\t0.28114\t0.63151\t\n",
      "4\t-228.93\t0.27750\t0.63510\t\n",
      "5\t-228.93\t0.27812\t0.63452\t\n",
      "6\t-228.93\t0.27801\t0.63461\t\n",
      "7\t-228.93\t0.27803\t0.63460\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.278031\t0.137515\n",
      "V(e)\t0.634598\t0.135991\n",
      "Vp\t0.912628\t0.058021\n",
      "V(G)/Vp\t0.304648\t0.147295\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "1.891030e-02\t-1.701866e-02\t\n",
      "-1.701866e-02\t1.849343e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_4.hsq].\n",
      "\n",
      "Analysis finished at 03:08:51 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.11 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:51 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_5\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_5\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_5.grm.id].\n",
      "502 IDs are read from [EUR_5.grm.id].\n",
      "Reading the GRM from [EUR_5.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_5.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.458011 0.477687\n",
      "logL: -236.395\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-236.05\t0.30743\t0.61024\t\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2\t-233.56\t0.02879\t0.87306\t\n",
      "3\t-232.16\t0.05084\t0.85548\t\n",
      "4\t-232.02\t0.06598\t0.84371\t\n",
      "5\t-231.96\t0.09756\t0.81983\t\n",
      "6\t-231.92\t0.09220\t0.82543\t\n",
      "7\t-231.92\t0.09339\t0.82426\t\n",
      "8\t-231.92\t0.09313\t0.82451\t\n",
      "9\t-231.92\t0.09319\t0.82446\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.093188\t0.118496\n",
      "V(e)\t0.824456\t0.127563\n",
      "Vp\t0.917644\t0.058010\n",
      "V(G)/Vp\t0.101552\t0.128760\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "1.404135e-02\t-1.347418e-02\t\n",
      "-1.347418e-02\t1.627221e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_5.hsq].\n",
      "\n",
      "Analysis finished at 03:08:51 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.12 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:51 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_6\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_6\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_6.grm.id].\n",
      "502 IDs are read from [EUR_6.grm.id].\n",
      "Reading the GRM from [EUR_6.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_6.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.455974 0.481474\n",
      "logL: -235.886\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-235.37\t0.28758\t0.62294\t\n",
      "2\t-231.92\t0.04276\t0.85186\t\n",
      "3\t-231.20\t0.07053\t0.82823\t\n",
      "4\t-230.75\t0.13738\t0.77321\t\n",
      "5\t-230.41\t0.12792\t0.78482\t\n",
      "6\t-230.41\t0.13127\t0.78162\t\n",
      "7\t-230.41\t0.13017\t0.78269\t\n",
      "8\t-230.41\t0.13054\t0.78233\t\n",
      "9\t-230.41\t0.13041\t0.78245\t\n",
      "10\t-230.41\t0.13045\t0.78241\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.130454\t0.093751\n",
      "V(e)\t0.782410\t0.103146\n",
      "Vp\t0.912864\t0.057748\n",
      "V(G)/Vp\t0.142906\t0.101855\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "8.789322e-03\t-8.046806e-03\t\n",
      "-8.046806e-03\t1.063910e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_6.hsq].\n",
      "\n",
      "Analysis finished at 03:08:52 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.15 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:52 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_7\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_7\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_7.grm.id].\n",
      "502 IDs are read from [EUR_7.grm.id].\n",
      "Reading the GRM from [EUR_7.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_7.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.458097 0.467732\n",
      "logL: -230.78\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-230.70\t0.38279\t0.53604\t\n",
      "2\t-230.06\t0.23233\t0.67717\t\n",
      "3\t-229.58\t0.25272\t0.66166\t\n",
      "4\t-229.56\t0.25104\t0.66335\t\n",
      "5\t-229.56\t0.25122\t0.66318\t\n",
      "6\t-229.56\t0.25120\t0.66320\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.251201\t0.134405\n",
      "V(e)\t0.663198\t0.133757\n",
      "Vp\t0.914400\t0.058132\n",
      "V(G)/Vp\t0.274717\t0.144024\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "1.806475e-02\t-1.628819e-02\t\n",
      "-1.628819e-02\t1.789094e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_7.hsq].\n",
      "\n",
      "Analysis finished at 03:08:52 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.12 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:52 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_8\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_8\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_8.grm.id].\n",
      "502 IDs are read from [EUR_8.grm.id].\n",
      "Reading the GRM from [EUR_8.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_8.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.457875 0.478743\n",
      "logL: -236.057\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-235.68\t0.31482\t0.60319\t\n",
      "2\t-233.21\t0.05450\t0.84703\t\n",
      "3\t-231.84\t0.07497\t0.83106\t\n",
      "4\t-231.72\t0.08897\t0.82042\t\n",
      "5\t-231.66\t0.11825\t0.79879\t\n",
      "6\t-231.63\t0.11370\t0.80352\t\n",
      "7\t-231.63\t0.11464\t0.80261\t\n",
      "8\t-231.63\t0.11445\t0.80279\t\n",
      "9\t-231.63\t0.11449\t0.80276\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.114492\t0.113860\n",
      "V(e)\t0.802756\t0.121876\n",
      "Vp\t0.917248\t0.058022\n",
      "V(G)/Vp\t0.124821\t0.123499\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "1.296406e-02\t-1.222563e-02\t\n",
      "-1.222563e-02\t1.485374e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_8.hsq].\n",
      "\n",
      "Analysis finished at 03:08:52 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.12 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:52 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_9\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_9\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_9.grm.id].\n",
      "502 IDs are read from [EUR_9.grm.id].\n",
      "Reading the GRM from [EUR_9.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_9.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.457276  0.46266\n",
      "logL: -226.874\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-226.86\t0.42145\t0.49471\t\n",
      "2\t-226.68\t0.34897\t0.56074\t\n",
      "3\t-226.55\t0.35709\t0.55423\t\n",
      "4\t-226.55\t0.35640\t0.55488\t\n",
      "5\t-226.55\t0.35647\t0.55482\t\n",
      "6\t-226.55\t0.35646\t0.55482\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.356462\t0.133133\n",
      "V(e)\t0.554823\t0.126359\n",
      "Vp\t0.911285\t0.058408\n",
      "V(G)/Vp\t0.391164\t0.139775\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "1.772428e-02\t-1.513976e-02\t\n",
      "-1.513976e-02\t1.596668e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_9.hsq].\n",
      "\n",
      "Analysis finished at 03:08:52 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.08 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:52 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_10\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_10\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_10.grm.id].\n",
      "502 IDs are read from [EUR_10.grm.id].\n",
      "Reading the GRM from [EUR_10.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_10.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Updated prior values:  0.45962 0.474689\n",
      "logL: -234.173\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-233.94\t0.35108\t0.57114\t\n",
      "2\t-232.54\t0.12879\t0.77858\t\n",
      "3\t-231.37\t0.13560\t0.77495\t\n",
      "4\t-231.35\t0.14029\t0.77247\t\n",
      "5\t-231.34\t0.15040\t0.76719\t\n",
      "6\t-231.34\t0.15008\t0.76755\t\n",
      "7\t-231.34\t0.15010\t0.76753\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.150095\t0.122991\n",
      "V(e)\t0.767531\t0.127696\n",
      "Vp\t0.917626\t0.058152\n",
      "V(G)/Vp\t0.163569\t0.132831\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "1.512671e-02\t-1.402569e-02\t\n",
      "-1.402569e-02\t1.630629e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_10.hsq].\n",
      "\n",
      "Analysis finished at 03:08:52 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.12 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:52 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_11\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_11\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_11.grm.id].\n",
      "502 IDs are read from [EUR_11.grm.id].\n",
      "Reading the GRM from [EUR_11.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_11.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.460105 0.480199\n",
      "logL: -234.97\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-234.55\t0.33770\t0.58505\t\n",
      "2\t-232.58\t0.10251\t0.80059\t\n",
      "3\t-231.16\t0.11391\t0.79359\t\n",
      "4\t-231.11\t0.12202\t0.78860\t\n",
      "5\t-231.09\t0.13986\t0.77768\t\n",
      "6\t-231.07\t0.13938\t0.77827\t\n",
      "7\t-231.07\t0.13942\t0.77823\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.139416\t0.107848\n",
      "V(e)\t0.778234\t0.113253\n",
      "Vp\t0.917650\t0.058236\n",
      "V(G)/Vp\t0.151928\t0.116228\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "1.163116e-02\t-1.053294e-02\t\n",
      "-1.053294e-02\t1.282621e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_11.hsq].\n",
      "\n",
      "Analysis finished at 03:08:52 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.12 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:52 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_12\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_12\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_12.grm.id].\n",
      "502 IDs are read from [EUR_12.grm.id].\n",
      "Reading the GRM from [EUR_12.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_12.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.458282 0.478942\n",
      "logL: -235.712\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-235.33\t0.32096\t0.59797\t\n",
      "2\t-232.99\t0.07013\t0.83189\t\n",
      "3\t-231.64\t0.08849\t0.81798\t\n",
      "4\t-231.54\t0.10111\t0.80865\t\n",
      "5\t-231.50\t0.12766\t0.78950\t\n",
      "6\t-231.47\t0.12411\t0.79318\t\n",
      "7\t-231.47\t0.12476\t0.79255\t\n",
      "8\t-231.47\t0.12465\t0.79266\t\n",
      "9\t-231.47\t0.12467\t0.79265\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.124666\t0.112822\n",
      "V(e)\t0.792645\t0.119982\n",
      "Vp\t0.917312\t0.058065\n",
      "V(G)/Vp\t0.135904\t0.122171\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "1.272887e-02\t-1.187654e-02\t\n",
      "-1.187654e-02\t1.439569e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_12.hsq].\n",
      "\n",
      "Analysis finished at 03:08:52 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.11 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:52 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_13\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_13\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_13.grm.id].\n",
      "502 IDs are read from [EUR_13.grm.id].\n",
      "Reading the GRM from [EUR_13.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_13.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.456358 0.473792\n",
      "logL: -230.891\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-230.66\t0.36857\t0.54912\t\n",
      "2\t-229.40\t0.20260\t0.70039\t\n",
      "3\t-228.61\t0.21319\t0.69260\t\n",
      "4\t-228.58\t0.23511\t0.67687\t\n",
      "5\t-228.56\t0.23219\t0.67977\t\n",
      "6\t-228.56\t0.23268\t0.67931\t\n",
      "7\t-228.56\t0.23260\t0.67939\t\n",
      "8\t-228.56\t0.23261\t0.67938\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.232614\t0.109749\n",
      "V(e)\t0.679377\t0.110581\n",
      "Vp\t0.911991\t0.058028\n",
      "V(G)/Vp\t0.255062\t0.117340\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "1.204485e-02\t-1.045287e-02\t\n",
      "-1.045287e-02\t1.222809e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_13.hsq].\n",
      "\n",
      "Analysis finished at 03:08:52 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.11 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:52 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_14\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_14\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_14.grm.id].\n",
      "502 IDs are read from [EUR_14.grm.id].\n",
      "Reading the GRM from [EUR_14.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_14.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.456104  0.49184\n",
      "logL: -239.813\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-238.77\t0.28492\t0.62975\t\n",
      "2\t-234.18\t0.19597\t0.71117\t\n",
      "3\t-232.54\t0.14838\t0.75896\t\n",
      "4\t-231.97\t0.12214\t0.78719\t\n",
      "5\t-231.76\t0.07515\t0.84059\t\n",
      "6\t-231.66\t0.08949\t0.82753\t\n",
      "7\t-231.65\t0.08573\t0.83127\t\n",
      "8\t-231.65\t0.08680\t0.83024\t\n",
      "9\t-231.65\t0.08650\t0.83053\t\n",
      "10\t-231.65\t0.08659\t0.83045\t\n",
      "11\t-231.65\t0.08656\t0.83047\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.086563\t0.090222\n",
      "V(e)\t0.830469\t0.102228\n",
      "Vp\t0.917032\t0.057987\n",
      "V(G)/Vp\t0.094394\t0.097965\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "8.140097e-03\t-7.614095e-03\t\n",
      "-7.614095e-03\t1.045056e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_14.hsq].\n",
      "\n",
      "Analysis finished at 03:08:52 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.14 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:52 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_15\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_15\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_15.grm.id].\n",
      "502 IDs are read from [EUR_15.grm.id].\n",
      "Reading the GRM from [EUR_15.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_15.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.460093 0.471215\n",
      "logL: -229.905\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-229.77\t0.40403\t0.52119\t\n",
      "2\t-229.25\t0.28323\t0.63245\t\n",
      "3\t-228.75\t0.28162\t0.63733\t\n",
      "4\t-228.75\t0.28151\t0.63745\t\n",
      "5\t-228.75\t0.28150\t0.63746\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.281504\t0.121263\n",
      "V(e)\t0.637457\t0.116513\n",
      "Vp\t0.918960\t0.059026\n",
      "V(G)/Vp\t0.306328\t0.126989\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "1.470481e-02\t-1.239805e-02\t\n",
      "-1.239805e-02\t1.357531e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_15.hsq].\n",
      "\n",
      "Analysis finished at 03:08:53 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.11 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:53 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_16\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_16\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_16.grm.id].\n",
      "502 IDs are read from [EUR_16.grm.id].\n",
      "Reading the GRM from [EUR_16.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_16.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values:  0.45745 0.472241\n",
      "logL: -230.99\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-230.81\t0.37263\t0.54639\t\n",
      "2\t-229.76\t0.20915\t0.69687\t\n",
      "3\t-229.05\t0.21845\t0.69000\t\n",
      "4\t-229.03\t0.23778\t0.67603\t\n",
      "5\t-229.01\t0.23550\t0.67829\t\n",
      "6\t-229.01\t0.23584\t0.67797\t\n",
      "7\t-229.01\t0.23579\t0.67802\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.235789\t0.116397\n",
      "V(e)\t0.678018\t0.116590\n",
      "Vp\t0.913808\t0.058160\n",
      "V(G)/Vp\t0.258030\t0.124350\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "1.354820e-02\t-1.187944e-02\t\n",
      "-1.187944e-02\t1.359326e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_16.hsq].\n",
      "\n",
      "Analysis finished at 03:08:53 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.10 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:53 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_17\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_17\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_17.grm.id].\n",
      "502 IDs are read from [EUR_17.grm.id].\n",
      "Reading the GRM from [EUR_17.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_17.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.460035 0.498312\n",
      "logL: -243.727\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-242.30\t0.24928\t0.66581\t\n",
      "2\t-236.37\t0.13488\t0.76936\t\n",
      "3\t-233.95\t0.07153\t0.83268\t\n",
      "4\t-232.99\t0.03532\t0.87168\t\n",
      "5\t-232.60\t0.00000\t0.91641\t(1 component(s) constrained)\n",
      "6\t-232.37\t0.00000\t0.91834\t(1 component(s) constrained)\n",
      "7\t-232.37\t0.00000\t0.91835\t(1 component(s) constrained)\n",
      "8\t-232.37\t0.00000\t0.91835\t(1 component(s) constrained)\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.000001\t0.082601\n",
      "V(e)\t0.918348\t0.100007\n",
      "Vp\t0.918349\t0.058024\n",
      "V(G)/Vp\t0.000001\t0.089946\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "6.822992e-03\t-6.728779e-03\t\n",
      "-6.728779e-03\t1.000132e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_17.hsq].\n",
      "\n",
      "Analysis finished at 03:08:53 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.13 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:53 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_18\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_18\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_18.grm.id].\n",
      "502 IDs are read from [EUR_18.grm.id].\n",
      "Reading the GRM from [EUR_18.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_18.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values:  0.45543 0.487758\n",
      "logL: -237.501\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-236.68\t0.30243\t0.61239\t\n",
      "2\t-232.92\t0.05532\t0.83921\t\n",
      "3\t-231.40\t0.08023\t0.82001\t\n",
      "4\t-231.13\t0.09715\t0.80758\t\n",
      "5\t-231.02\t0.13191\t0.78334\t\n",
      "6\t-230.95\t0.12437\t0.79118\t\n",
      "7\t-230.95\t0.12651\t0.78913\t\n",
      "8\t-230.95\t0.12592\t0.78970\t\n",
      "9\t-230.95\t0.12609\t0.78954\t\n",
      "10\t-230.95\t0.12604\t0.78959\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.126041\t0.094407\n",
      "V(e)\t0.789585\t0.103441\n",
      "Vp\t0.915626\t0.057954\n",
      "V(G)/Vp\t0.137656\t0.102219\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "8.912597e-03\t-8.126994e-03\t\n",
      "-8.126994e-03\t1.070011e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_18.hsq].\n",
      "\n",
      "Analysis finished at 03:08:53 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.13 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:53 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_19\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_19\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_19.grm.id].\n",
      "502 IDs are read from [EUR_19.grm.id].\n",
      "Reading the GRM from [EUR_19.grm.bin].\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GRM for 502 individuals are included from [EUR_19.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.459716 0.456966\n",
      "logL: -221.218\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-221.21\t0.47391\t0.44414\t\n",
      "2\t-221.18\t0.50701\t0.41457\t\n",
      "3\t-221.14\t0.51223\t0.41039\t\n",
      "4\t-221.14\t0.51298\t0.40976\t\n",
      "5\t-221.14\t0.51308\t0.40967\t\n",
      "6\t-221.14\t0.51310\t0.40966\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.513099\t0.135268\n",
      "V(e)\t0.409658\t0.113775\n",
      "Vp\t0.922758\t0.061812\n",
      "V(G)/Vp\t0.556050\t0.129945\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "1.829732e-02\t-1.371070e-02\t\n",
      "-1.371070e-02\t1.294482e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_19.hsq].\n",
      "\n",
      "Analysis finished at 03:08:53 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.12 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:53 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_20\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_20\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_20.grm.id].\n",
      "502 IDs are read from [EUR_20.grm.id].\n",
      "Reading the GRM from [EUR_20.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_20.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.459954 0.487954\n",
      "logL: -238.464\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-237.68\t0.31459\t0.60783\t\n",
      "2\t-234.51\t0.03012\t0.86421\t\n",
      "3\t-232.14\t0.04554\t0.85574\t\n",
      "4\t-232.00\t0.05657\t0.84976\t\n",
      "5\t-231.94\t0.08074\t0.83696\t\n",
      "6\t-231.89\t0.07929\t0.83875\t\n",
      "7\t-231.89\t0.07946\t0.83858\t\n",
      "8\t-231.89\t0.07944\t0.83860\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.079442\t0.094102\n",
      "V(e)\t0.838604\t0.105489\n",
      "Vp\t0.918046\t0.058070\n",
      "V(G)/Vp\t0.086534\t0.102097\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "8.855115e-03\t-8.305469e-03\t\n",
      "-8.305469e-03\t1.112790e-02\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_20.hsq].\n",
      "\n",
      "Analysis finished at 03:08:53 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.11 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:53 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_21\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_21\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_21.grm.id].\n",
      "502 IDs are read from [EUR_21.grm.id].\n",
      "Reading the GRM from [EUR_21.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_21.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.457824 0.514268\n",
      "logL: -246.464\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-243.72\t0.26347\t0.65371\t\n",
      "2\t-236.64\t0.16372\t0.73994\t\n",
      "3\t-233.88\t0.11005\t0.79316\t\n",
      "4\t-232.84\t0.07999\t0.82614\t\n",
      "5\t-232.44\t0.06261\t0.84671\t\n",
      "6\t-232.29\t0.03009\t0.88761\t\n",
      "7\t-232.20\t0.03785\t0.88083\t\n",
      "8\t-232.19\t0.03626\t0.88241\t\n",
      "9\t-232.19\t0.03661\t0.88208\t\n",
      "10\t-232.19\t0.03653\t0.88215\t\n",
      "11\t-232.19\t0.03655\t0.88213\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.036550\t0.071907\n",
      "V(e)\t0.882134\t0.089395\n",
      "Vp\t0.918684\t0.058078\n",
      "V(G)/Vp\t0.039785\t0.078146\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "5.170635e-03\t-4.894464e-03\t\n",
      "-4.894464e-03\t7.991387e-03\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_21.hsq].\n",
      "\n",
      "Analysis finished at 03:08:53 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.14 sec.\n",
      "*******************************************************************\n",
      "* Genome-wide Complex Trait Analysis (GCTA)\n",
      "* version v1.94.1 Linux\n",
      "* Built at Nov 15 2022 21:14:25, by GCC 8.5\n",
      "* (C) 2010-present, Yang Lab, Westlake University\n",
      "* Please report bugs to Jian Yang <jian.yang@westlake.edu.cn>\n",
      "*******************************************************************\n",
      "Analysis started at 03:08:53 UTC on Mon Mar 04 2024.\n",
      "Hostname: dsmlp-jupyter-hex002\n",
      "\n",
      "Accepted options:\n",
      "--grm EUR_22\n",
      "--pheno kgvcf_ldl.phen\n",
      "--reml\n",
      "--out EUR_22\n",
      "--thread-num 10\n",
      "\n",
      "Note: the program will be running on 10 threads.\n",
      "\n",
      "Reading IDs of the GRM from [EUR_22.grm.id].\n",
      "502 IDs are read from [EUR_22.grm.id].\n",
      "Reading the GRM from [EUR_22.grm.bin].\n",
      "GRM for 502 individuals are included from [EUR_22.grm.bin].\n",
      "Reading phenotypes from [kgvcf_ldl.phen].\n",
      "Non-missing phenotypes of 2504 individuals are included from [kgvcf_ldl.phen].\n",
      "\n",
      "502 individuals are in common in these files.\n",
      "\n",
      "Performing  REML analysis ... (Note: may take hours depending on sample size).\n",
      "502 observations, 1 fixed effect(s), and 2 variance component(s)(including residual variance).\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.458026 0.504544\n",
      "logL: -243.528\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(G)\tV(e)\t\n",
      "1\t-241.62\t0.27857\t0.63929\t\n",
      "2\t-235.85\t0.18085\t0.72477\t\n",
      "3\t-233.46\t0.12760\t0.77724\t\n",
      "4\t-232.54\t0.09785\t0.80933\t\n",
      "5\t-232.20\t0.04362\t0.87202\t\n",
      "6\t-231.99\t0.05695\t0.86075\t\n",
      "7\t-231.98\t0.05454\t0.86323\t\n",
      "8\t-231.98\t0.05505\t0.86273\t\n",
      "9\t-231.98\t0.05494\t0.86284\t\n",
      "10\t-231.98\t0.05496\t0.86282\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Calculating the logLikelihood for the reduced model ...\n",
      "(variance component 1 is dropped from the model)\n",
      "Calculating prior values of variance components by EM-REML ...\n",
      "Updated prior values: 0.91870\n",
      "logL: -232.36696\n",
      "Running AI-REML algorithm ...\n",
      "Iter.\tlogL\tV(e)\t\n",
      "1\t-232.37\t0.91870\t\n",
      "Log-likelihood ratio converged.\n",
      "\n",
      "Summary result of REML analysis:\n",
      "Source\tVariance\tSE\n",
      "V(G)\t0.054964\t0.074466\n",
      "V(e)\t0.862815\t0.090774\n",
      "Vp\t0.917780\t0.058016\n",
      "V(G)/Vp\t0.059888\t0.080931\n",
      "\n",
      "Sampling variance/covariance of the estimates of variance components:\n",
      "5.545172e-03\t-5.209616e-03\t\n",
      "-5.209616e-03\t8.239866e-03\t\n",
      "\n",
      "Summary result of REML analysis has been saved in the file [EUR_22.hsq].\n",
      "\n",
      "Analysis finished at 03:08:53 UTC on Mon Mar 04 2024\n",
      "Overall computational time: 0.12 sec.\n"
     ]
    }
   ],
   "source": [
    "for anc in ['EUR']:\n",
    "    for i in range(1,23):\n",
    "        os.system(f\"./gcta/gcta-1.94.1 --grm EUR_{i} --pheno kgvcf_ldl.phen --reml --out EUR_{i} --thread-num 10\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "807d1a25",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_1_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_1\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_1_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "6468094 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_1_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99987.\n",
      "5933277 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "534817 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_1_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_2_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_2\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_2_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "7081600 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_2_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999873.\n",
      "6503484 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "578116 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_2_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_3_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_3\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_3_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "5832276 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_3_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999863.\n",
      "5339850 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "492426 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_3_LDL.assoc.linear ... 10111213141516171819202122232425262728293031323333343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_4_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_4\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_4_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "5732585 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_4_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999852.\n",
      "5224203 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "508382 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_4_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_5_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_5\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_5_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "5265763 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_5_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999868.\n",
      "4828018 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "437745 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_5_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_6_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_6\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_6_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "5024119 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_6_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999837.\n",
      "4545040 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "479079 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing linear model association results to EUR_6_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_7_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_7\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_7_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "4716715 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_7_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999866.\n",
      "4303432 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "413283 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_7_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_8_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_8\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_8_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "4597105 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_8_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999872.\n",
      "4219151 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "377954 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_8_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_9_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_9\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_9_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "3560687 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_9_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999866.\n",
      "3259429 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "301258 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_9_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_10_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_10\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_10_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "3992219 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_10_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99986.\n",
      "3632805 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "359414 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_10_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_11_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_11\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_11_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "4045628 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_11_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999862.\n",
      "3696225 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "349403 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_11_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_12_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_12\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_12_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "3868428 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_12_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999855.\n",
      "3531725 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "336703 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing linear model association results to EUR_12_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_13_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_13\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_13_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "2857916 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_13_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999842.\n",
      "2599751 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "258165 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_13_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_14_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_14\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_14_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "2655067 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_14_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999869.\n",
      "2426126 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "228941 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_14_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_15_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_15\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_15_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "2424689 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_15_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999863.\n",
      "2224980 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "199709 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_15_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_16_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_16\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_16_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "2697949 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_16_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999882.\n",
      "2484415 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "213534 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_16_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_17_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_17\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_17_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "2329288 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_17_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999877.\n",
      "2137109 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "192179 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_17_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_18_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_18\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_18_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "2267185 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_18_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999855.\n",
      "2069950 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "197235 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing linear model association results to EUR_18_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_19_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_19\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_19_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "1832506 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_19_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999863.\n",
      "1664784 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "167722 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_19_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_20_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_20\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_20_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "1812841 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_20_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999882.\n",
      "1659551 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "153290 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_20_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_21_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_21\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_21_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "1105538 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_21_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999831.\n",
      "1003661 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "101877 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_21_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR_22_LDL.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile EUR_22\n",
      "  --linear\n",
      "  --maf 0.05\n",
      "  --out EUR_22_LDL\n",
      "  --pheno /home/hex002/public/ps4/kgvcf_ldl.phen\n",
      "\n",
      "385394 MB RAM detected; reserving 192697 MB for main workspace.\n",
      "1103547 variants loaded from .bim file.\n",
      "502 people (0 males, 0 females, 502 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to EUR_22_LDL.nosex .\n",
      "502 phenotype values present after --pheno.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 502 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999881.\n",
      "1004572 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "98975 variants and 502 people pass filters and QC.\n",
      "Phenotype data is quantitative.\n",
      "Writing linear model association results to EUR_22_LDL.assoc.linear ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "for anc in ['EUR']:\n",
    "    for i in range(1,23):\n",
    "        os.system(f\"plink --bfile EUR_{i} \\\n",
    "      --pheno ~/public/ps4/kgvcf_ldl.phen \\\n",
    "      --linear \\\n",
    "      --allow-no-sex \\\n",
    "      --maf 0.05 \\\n",
    "      --out  EUR_{i}_LDL\")\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
